enterprise/cmd/frontend/internal/dotcom/productsubscription/service_account.go

```diff
@@ -0,0 +1,27 @@
+package productsubscription
+
+import (
+	"context"
+
+	"github.com/sourcegraph/sourcegraph/internal/auth"
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/featureflag"
+)
+
+// featureFlagProductSubscriptionsServiceAccount is a feature flag that should be
+// set on service accounts that can read product subscriptions.
+const featureFlagProductSubscriptionsServiceAccount = "product-subscriptions-service-account"
+
+// 🚨 SECURITY: Use this to check if access to a subscription query or mutation
+// is authorized for service accounts, the owning user, and site admins.
+func serviceAccountOrOwnerOrSiteAdmin(ctx context.Context, db database.DB, ownerUserID *int32) error {
+	if featureflag.FromContext(ctx).GetBoolOr(featureFlagProductSubscriptionsServiceAccount, false) {
+		return nil
+	}
+
+	if ownerUserID != nil {
+		return auth.CheckSiteAdminOrSameUser(ctx, db, *ownerUserID)
+	}
```

Comment:
If owner is not nil, should it be enough to check if it's the same user?
```suggestion
	if ownerUserID != nil {
		return auth.CheckSameUser(ctx, *ownerUserID)
	}
```

Reply:
Reverted this - the actor can either be an owner, or a site admin, so `CheckSiteAdminOrSameUser` is the right check